File: day_1/main.py
data = open("input.txt").read()

elves = data.split("\n\n")

max_load = 0
elves_load = []

for elf in elves:
    load = sum([int(load) for load in elf.split()])
    elves_load.append(load)
    max_load = max(load, max_load)

print(max_load)
assert max_load == 72070

top_max_load = sum(list(reversed(sorted(elves_load)))[:3])
print(top_max_load)
assert top_max_load == 211805

File: day_1/main.go
package main

import (
	"fmt"
	"os"
	"sort"
	"strconv"
	"strings"
)

func main() {
	data, _ := os.ReadFile("input.txt")

	elves := strings.Split(string(data), "\n\n")

	maxLoad := 0
	elvesLoad := make([]int, 0)

	for _, elf := range elves {
		sumLoad := 0
		for _, value := range strings.Split(elf, "\n") {
			load, _ := strconv.Atoi(value)
			sumLoad += load
		}
		maxLoad = max(sumLoad, maxLoad)
		elvesLoad = append(elvesLoad, sumLoad)

	}

	fmt.Println(maxLoad)

	sort.Slice(elvesLoad, func(a, b int) bool { return elvesLoad[a] > elvesLoad[b] })

	topMaxLoad := 0
	for _, load := range elvesLoad[:3] {
		topMaxLoad += load
	}
	fmt.Println(topMaxLoad)
}

func max(x, y int) int {
	if x < y {
		return y
	}
	return x
}
